adminSiteServer/chartConfigR2Helpers.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    GRAPHER_CONFIG_R2_BUCKET,
+    GRAPHER_CONFIG_R2_BUCKET_PATH,
+    R2_ACCESS_KEY_ID,
+    R2_ENDPOINT,
+    R2_REGION,
+    R2_SECRET_ACCESS_KEY,
+} from "../settings/serverSettings.js"
+import {
+    DeleteObjectCommand,
+    DeleteObjectCommandInput,
+    PutObjectCommand,
+    PutObjectCommandInput,
+    S3Client,
+} from "@aws-sdk/client-s3"
+import { Base64String, JsonError } from "@ourworldindata/utils"
+import { logErrorAndMaybeSendToBugsnag } from "../serverUtils/errorLog.js"
+import { createHash } from "crypto"
+
+export function getMd5HashBase64(data: string): Base64String {
+    // I would have liked to create a function in utils that can compute a varienty of hashes
+    // in both the browser, CF workers and node but unfortunately this isn't easily possible
+    // for md5 - so here we just special case for md5, node and base64 encoding for now.
```

Comment:
Yeah that's what I tried first and with SHA hashes it would have mostly worked as you say. But while the S3 client and api responses can handle SHA hashes on many calls, they don't do that on the list objects call where they only support md5. This means that you can get up to 1000 md5 hashes for files on R2 in one call but would have to perform a single http call per file if you want to know any of the SHA hashes. Since the list operation is the one that motivates the whole hashing in the first place, I switched to MD5 after I had already implemeneted SHA everywhere.

In keeping with your suggestion to move the hex/base64 utils to serverUtils I now also moved the md5 hashing function over to serverUtils